docs/adr/0035-refactor-payments-frontend.md

```diff
@@ -0,0 +1,101 @@
+
+Finally, the frontend includes some business and application logic, which has made components and parts of the app unnecessarily complex. As a result the frontend is less flexible, and harder to change and maintain. To solve this, the Subscription Platform team wants to move all business and application logic to the backend.
+
+To address the problems briefly discussed above, and to meet current and future relying party and feature requests, refactoring of the existing Subscription Platform is necessary.
+
+## Decision Drivers
+
+- Faster development - This applies to both the refactoring necessary for SubPlat 3.0, as well as future relying party and feature requests
+- Easier routing - The SubPlat teams wants to perform more advanced routing to allow features such as, Relying Party specific checkout pages, A/B testing, etc,
+- GraphQL support - Improve app performance and responsivness by reducing the number of network requests
+- Reuse existing components - We have received generally favourable reviews of our existing frontend look and feel, and want to keep the appearance the same
+
+## Considered Options
+
+- A. React Framework - Next.js
+- B. React Framework - Gatsby.js
+- C. React, Typescript, without Redux
+- D. React, Typescript, with Redux
+- E. React Framework - Remix
+
+## Decision Outcome
+
+Chosen option: "A. React Framework - Next.js", because a significant amount of refactoring is necessary to accomplish the Subscription Platforms goals, now would be a good time to adopt a React framework and start from a mostly clean slate, since it partially or completely meets all of the decision drivers, especially the easier routing and faster development drivers.
+
+Regarding why Next.js is favored over Gatsby.js, before completing the ADR, the subscription platform team went through a prototyping project and found the two frameworks to be remarkably similar in capabilities. However due to Next.js's popularity, use by other teams in Mozilla and a team vote, the decision was made to pick Next.js.
+
+## Pros and Cons of the Options
+
+### A. React Framework - Next.js
+
+Use a React framework, Next.js.
+
+- Good, because it provides many benefits out of the box, without compromising flexibility, that would require additional libraries otherwise. (Routing, rendering options, image optimization, etc.)
+- Good, because of its routing simplicity and flexibility
+- Good, because of its built in tooling, which reduces existing bespoke tooling currently used
+- Good, because it is easy to learn the basics
+- Good, because it's parent company has other tools that work well with Next.js and enhance its capabilities. (Turbopack, SWR, etc.)
+- Good, because of its large community support and popularity
+- Good, because it is used by other teams in Mozilla
+- Good, because it no longer uses Redux for state management
+- Bad, because it adds another layer on top of React
```

Comment:
And another dependency.

Also, are we going to use most of the features NextJS offers? It could be a con that it's heavier than vanilla React and we may not utilize all of its features, but you would know better than me.